### PR TITLE
use generator expressions for configuration specific tests

### PR DESCRIPTION
**Pull request by dirk-thomas**
Debug: http://ci.ros2.org/job/ci_windows/994/
Release: http://ci.ros2.org/job/ci_windows/995/

### CODE REVIEW

File: intra_process_demo/CMakeLists.txt [hunk 1]
```diff
@@ -127,18 +116,22 @@ if(AMENT_ENABLE_TESTING)
...
         @ONLY
       )
+      file(GENERATE
+        OUTPUT "test_${exe_list_underscore}${target_suffix}_$<CONFIGURATION>.py"
+        INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}.py.configured"
+      )
 
       ament_add_nose_test(test_demo_${exe_list_underscore}${target_suffix}
-        "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}.py"
+        "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}_$<CONFIGURATION>.py"
```
Comment (tfoote) [inline intra_process_demo/CMakeLists.txt]: Reading the docs CONFIGURATION is deprecated and CONFIG should be used instead.

Comment (dirk-thomas) [inline intra_process_demo/CMakeLists.txt]: Can you please point to the docs.

Comment (wjwwood) [inline intra_process_demo/CMakeLists.txt]: https://cmake.org/cmake/help/v3.0/manual/cmake-generator-expressions.7.html#informational-expressions

Comment (tfoote) [inline intra_process_demo/CMakeLists.txt]: https://cmake.org/cmake/help/v3.0/manual/cmake-generator-expressions.7.html and theres an odd reference in 2.8.12 https://cmake.org/cmake/help/v2.8.12/cmake.html#section_Generators which is a little ambiguous

Comment (dirk-thomas) [inline intra_process_demo/CMakeLists.txt]: We are using 2.8.12 and as far as I can see it doesn't have that option: https://cmake.org/cmake/help/v2.8.12/cmake.html#command:add_compile_options

`$<CONFIGURATION>` evaluates to the configuration name.
`$<CONFIG:cfg>` evaluates to a boolean ('1' if config is "cfg", else '0').

Comment (tfoote) [inline intra_process_demo/CMakeLists.txt]: That's really stupid of them to deprecate a value and move to a new value that had a completely different meaning in their last major version.

**Comment by tfoote** [comment]
+1

**Comment by wjwwood** [comment]
@dirk-thomas I think the +1 covers the connected pr's? Maybe @tfoote disagrees, but I think these are ready for merge.

**Comment by tfoote** [comment]
Yeah, all 3 are good to merge.

**Pull request merged**